tests/testthat/test-add_annotation.R

```diff
@@ -17,6 +17,31 @@ test_that("add_CellLine_annotation works", {
   expect_true(all(dt_annotated$subtype == "unknown"))
 })
 
+
+test_that("add_CellLine_annotation works with custom annotation in external file", {
+  dt_unknown <- data.table::data.table(clid = "cl_id")
+  
+  # create custom annotations
+  temp_path <- tempfile(pattern = "file")
+  custom_annotation <- data.table::data.table(cell_line_identifier = "cl_id",
+                                              cell_line_name = "custom_name",
+                                              primary_tissue = "custom_tissue",
+                                              doubling_time = 99,
+                                              parental_identifier = "some text",
+                                              subtype = "random")
+  data.table::fwrite(custom_annotation,
+                     temp_path)
+  Sys.setenv(gDR_cellline_annotation = temp_path)
+  dt_unknown_annotated <- purrr::quietly(add_CellLine_annotation)(dt_unknown)$result
+  expect_equal(unname(dt_unknown_annotated), unname(custom_annotation))
+  
+  # restore default
+  Sys.setenv(gDR_cellline_annotation = "")
+  dt_unknown_annotated <- purrr::quietly(add_CellLine_annotation)(dt_unknown)$result
+  expect_equal(dt_unknown_annotated$clid, custom_annotation$cell_line_identifier)
+})
```

Comment:
Please, add some tests that show validation of annotation (scenario when `externalSource` is not valid annotation file)

Reply:
done

Reply:
✔️